src/openapi/streaming/subscription-queue.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Queue (FIFO) of pending subscription actions. Allows for easier abstraction of managing list of pending actions.
+ * In addition to standard queue implementation, provides:
+ *  - merging consecutive duplicated actions
+ *  - merging of uneccessary combinations: ACTION_SUBSCRIBE action before ACTION_UNSUBSCRIBE, is merged to ACTION_UNSUBSCRIBE.
+ *
+ * @module saxo/openapi/streaming/subscription queue
+ * @ignore
+ */
+
+import {
+	ACTION_SUBSCRIBE,
+	ACTION_UNSUBSCRIBE
+} from './subscription-actions';
+
+/**
+ * Queue (FIFO) for storing pending subscription actions.
+ * @param maxSize {Number} - Maximum queue size. Defaults to 2, which works best for current needs.
+ * @constructor
+ */
+function SubscriptionQueue(maxSize = 2) {
+	this.queue = [];
```

Comment:
I'd rename this to `items` or `array`, it's not a queue, but the wrapping class is